src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -356,8 +375,10 @@ private void WriteStreamReferenceProperty(string propertyName, ODataStreamRefere
         /// Writes a Null property.
         /// </summary>
         /// <param name="property">The property to write out.</param>
+        /// <param name="omitNullValues">Whether to omit the writing of this null property value.</param>
         private void WriteNullProperty(
-            ODataProperty property)
+            ODataProperty property,
+            bool omitNullValues)

```

Comment:
See above.  I would think that you can handle all cases of null and omit=null in WriteProperty, and never get to WriteNullProperty for null properties in the omit nulls case.

Reply:
For all cases of null and omit=null, code block around L392 is needed for <V7. Am I missing anything?

---
In reply to: [186562291](https://github.com/OData/odata.net/pull/1106#discussion_r186562291) [](ancestors = 186562291)

Reply:
You only call WriteNullProperty in one place, and you have an optimization before calling writeNullProperty that, if the property is null and omitNullValues is true, you return.  So this will only ever be called if it is a null value that you really want to write.

---
In reply to: [187208933](https://github.com/OData/odata.net/pull/1106#discussion_r187208933) [](ancestors = 187208933,186562291)

Reply:
omitNullVallues could be either true or false and needs to be passed in. See related comment in caller.

---
In reply to: [198079167](https://github.com/OData/odata.net/pull/1106#discussion_r198079167) [](ancestors = 198079167,187208933,186562291)

Reply:
see comment in caller. I don't see any reason why we can't check this case before calling the method and would have to pass it in.

---
In reply to: [209788338](https://github.com/OData/odata.net/pull/1106#discussion_r209788338) [](ancestors = 209788338,198079167,187208933,186562291)

Reply:
see response in the caller.

---
In reply to: [213515321](https://github.com/OData/odata.net/pull/1106#discussion_r213515321) [](ancestors = 213515321,209788338,198079167,187208933,186562291)